src/functions.jl

```diff
@@ -146,6 +146,78 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     return NamedDimsArray{newL}(data)
 end
 
+################################################
+# cat, vcat, hcat
+
+function Base.cat(a::NamedDimsArray{L}; dims) where L
+    newL = expand_dimnames(L, dims)
+    numerical_dims = dim(newL, dims)
+    data = Base.cat(parent(a); dims=numerical_dims)
+    newL = unify_names_longest(newL, dimnames(data))
+    return NamedDimsArray{newL}(data)
```

Comment:
reusing the name `newL` for variables of different types (in this case tuples of different lengths),
leads to type-instability.

I am not sure if there is an easier way to see this than using Cthulu.jl,
and viewing with `[w]arn` since it is a kwarg and so `@code_warntype` will only show the keyword method being called.
I guess you could rewrite it not to use a kwarg just for testing this out.